File: packages/helm-runtime-sdk/node/browser.go
// Package helmruntimenode embeds the Node runtime SDK's browser build so the
// daemon can serve it under /sdk/v1/ (docs/design/04-packages.md §8,
// docs/decisions.md M6 Q14). The JavaScript is the package; this file only
// carries it into the Go binary, and npm never publishes it ("files": ["src"]).
package helmruntimenode

import "embed"

// Browser holds the files a page imports: browser.js and what it imports.
// proxy.js and index.js are for Node and are not served.
//
//go:embed src/browser.js src/generated.js src/transport.js src/theme.js
var Browser embed.FS

// BrowserFiles are Browser's files, by the name they are served under
// /sdk/v1/runtime/.
var BrowserFiles = map[string]string{
	"browser.js":   "src/browser.js",
	"generated.js": "src/generated.js",
	"transport.js": "src/transport.js",
	"theme.js":     "src/theme.js",
}

File: packages/helm-runtime-sdk/node/src/browser.js
// helmstudio's runtime SDK for a studio's page: the same generated client as
// the Node package, over fetch, with no token. The studio's server mounts the
// runtime SDK's proxy at /helm/, which adds the token (docs/decisions.md M6
// Q10), so a page never holds one.
//
//   <script type="module">
//     import { connect, themeBridge } from "/helm/sdk/v1/helm-runtime.js";
//     themeBridge();
//     window.helm = connect();
//   </script>
//
// Served by the daemon as /sdk/v1/helm-runtime.js. No Node built-ins.

import { API_VERSION, Client } from "./generated.js";
import { HelmError, Transport } from "./transport.js";
import { applyTheme, themeBridge } from "./theme.js";

export { API_VERSION, Client, HelmError, Transport, applyTheme, themeBridge };

// connect returns a client for the platform API behind the page's proxy.
export function connect({ base = "/helm/api/v1", fetch } = {}) {
  return new Client(new Transport(base, undefined, { fetch }));
}
